### ISSUE TITLE
TypeAlias with ParamSpec isn't specialized correctly

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6826]
The following code should type check without error, but it doesn't.

```python
from typing import Callable, Concatenate, ParamSpec, TypeAlias, TypeVar, assert_type

P = ParamSpec("P")
R = TypeVar("R")

MyCallable: TypeAlias = Callable[Concatenate[int, P], R]


def func1(p1: MyCallable[[str, str], None]):
    assert_type(p1, Callable[[int, str, str], None])
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.344, which I just published. It will also be included in a future release of pylance.
